### DISCUSSION TITLE
Is there any bug or restriction that I am unable to set the revision in ArgoCD UI.

### DISCUSSION THREAD

**Discussion by mdinhvan** [argoproj/argo-cd#17723] [Q&A] [answered]
I am trying to perform as usual operations is updating the revision in ArgoCD UI (click Edit and set the revision I want), however currently using v2.10.5, it will automatically sync immediately to the latest ver.

**Comment by jgwest** [comment, marked as answer]
I'm not aware of any related bugs, and a quick search through Issues did not reveal any obvious candidates. If anyone is aware, feel free to reply.

You could try using `kubectl edit application/(resource)` to modify the revision directly via `kubectl`, rather than using Argo CD UI (to determine if it's a UI issue), or using Argo CD CLI.
- If you're able to consistently reproduce via UI, but not through CLI, AND you are able to provide simple reproduction steps, it would be worth opening a bug.

One thing to consider is: if you are using the app-of-apps pattern, that is, you have your Argo CD `Application` CRs defined within your GitOps repository, then Argo CD would revert any changes that are not defined in Git (as is its remit).

**Reply by mdinhvan** [reply]
Thanks much for your reply Igwest.

I will try to use another way as your suggestion like using `kubectl` instead.

**Comment by ivankovnatsky** [comment]
I'm also experiencing this, getting this error in CLI:

```console
at Nov 07 12:45 ❯ argocd --grpc-web app set argo/shared-use1-devops-cluster-atlantis --revision feature/redacted-lower-case-name-with-hyphens --source-position 1
FATA[0001] rpc error: code = InvalidArgument desc = application spec for shared-use1-devops-cluster-atlantis is invalid: InvalidSpecError: Unable to generate manifests in /: rpc error: code = Unknown desc = invalid revision 'feature/redacted-lower-case-name-with-hyphens': improper constraint: feature/redacted-lower-case-name-with-hyphens 
☸ shared-use1-devops-cluster (internal-apps) in ~ 
at Nov 07 12:45 ❯ 
```

Tried both UI and directly updating in-cluster, no way. Sync policy is manual.

**Reply by ivankovnatsky** [reply]
I figured it's because that App is managed by ApplicationSet.
